wordpress-mobile/WordPress-iOS
# [Gutenberg] Autosave

This PR implements the Gutenberg autosave mechanism replacing the previously implemented client-side autosave.

**Update:** The new logic implemented is described on https://github.com/wordpress-mobile/WordPress-iOS/pull/12489#issuecomment-535434070
**Updated Gutenberg PR**: https://github.com/WordPress/gutenberg/pull/17548

- `gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1351
- `gutenberg` PR: https://github.com/WordPress/gutenberg/pull/17329
- `WPAndroid` PR: https://github.com/wordpress-mobile/WordPress-Android/pull/10483
- Original issue (Android): https://github.com/wordpress-mobile/WordPress-Android/issues/10415

**To test (updated):**
- **Run metro server** on `gutenberg-mobile:develop`
- Create/open a Gutenberg post.
- Write some content.
  - You can check if `editorDidAutosave()` is invoked on every content change.
  - Check that `debouncerCallback` is invoked after 1 second of inactivity.
  - Check that `debouncerCallback` is invoked after typing 50 characters without break.
- Kill the app and restart it again.
- Check that the post was automatically saved.

Update release notes:
